runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -36,8 +40,1040 @@ void J9::ARM64::AheadOfTimeCompile::processRelocations()
+         uint16_t symbolID = comp->getSymbolValidationManager()->getIDFromSymbol(static_cast<void *>(symbol));
+         TR_ASSERT_FATAL(symbolID, "symbolID should exist!\n");
+
+         cursor -= sizeof(TR_RelocationRecordBinaryTemplate);
+
+         TR_RelocationRecordResolvedTrampolinesBinaryTemplate *binaryTemplate =
+               reinterpret_cast<TR_RelocationRecordResolvedTrampolinesBinaryTemplate *>(cursor);
+
+         binaryTemplate->_symbolID = symbolID;
+
+         cursor += sizeof(TR_RelocationRecordResolvedTrampolinesBinaryTemplate);
+         }
+         break;
+
+      case TR_J2IVirtualThunkPointer:
+         {
+         auto info = (TR_RelocationRecordInformation*)relocation->getTargetAddress();
+
+         *(uintptrj_t *)cursor = (uintptrj_t)info->data2; // inlined site index
+         cursor += SIZEPOINTER;
+
+         *(uintptrj_t *)cursor = (uintptrj_t)info->data1; // constantPool
+         cursor += SIZEPOINTER;
+
+         *(uintptrj_t *)cursor = (uintptrj_t)info->data3; // offset to J2I virtual thunk pointer
+         cursor += SIZEPOINTER;
+         }
+         break;
+
+      case TR_CheckMethodExit:
+         {
+         *(uint64_t *) cursor = (uint64_t) (uintptrj_t) relocation->getTargetAddress();
+         cursor += SIZEPOINTER;
+         }
+         break;
+     case TR_J2IThunks:
+         {
+         TR::Node *node = (TR::Node*)relocation->getTargetAddress();
+         TR::SymbolReference *symRef = node->getSymbolReference();
+
+         *(uintptrj_t *)cursor = (uintptrj_t)node->getInlinedSiteIndex();
+         cursor += SIZEPOINTER;
+
+         *(uintptrj_t *)cursor = (uintptrj_t)symRef->getOwningMethod(comp)->constantPool(); // cp address
+         cursor += SIZEPOINTER;
+
+
+         *(uintptrj_t *)cursor = (uintptrj_t)symRef->getCPIndex(); // cp index
+         cursor += SIZEPOINTER;
+
+         break;
+         }
+      case TR_RamMethodSequence:
+      case TR_RamMethodSequenceReg:
+         {
+         *(uint64_t *) cursor = relocation->getTargetAddress() ?
+            (uint64_t)((uint8_t *) relocation->getTargetAddress() - aotMethodCodeStart) : 0x0;
+         cursor += SIZEPOINTER;
+         }
+         break;
+
+      case TR_MethodPointer:
+         {
+         TR::Node *aconstNode = (TR::Node *) relocation->getTargetAddress();
+         uintptrj_t inlinedSiteIndex = (uintptrj_t)aconstNode->getInlinedSiteIndex();
+         *(uintptrj_t *)cursor = inlinedSiteIndex;
+         cursor += SIZEPOINTER;
+
+         TR_OpaqueMethodBlock *j9method = (TR_OpaqueMethodBlock *) aconstNode->getAddress();
+         TR_OpaqueClassBlock *j9class = fej9->getClassFromMethodBlock(j9method);
+
+         uintptrj_t classChainOffsetInSharedCache = sharedCache->getClassChainOffsetOfIdentifyingLoaderForClazzInSharedCache(j9class);
+         *(uintptrj_t *)cursor = classChainOffsetInSharedCache;
+         cursor += SIZEPOINTER;
+
+         cursor = emitClassChainOffset(cursor, j9class);
```

Comment:
I think you need to do `self()->emitClassChainOffset(cursor, j9class);`

Reply:
I changed all the calls to `emitClassChainOffset` to use `self()`.